### DISCUSSION TITLE
bash equivalent of MAILCHECK and MAILPATH (checking for mail in /var/mail)

### DISCUSSION THREAD

**Discussion by valankar** [fish-shell/fish-shell#9937] [Q&A] [answered]
Hi,

Fish is great. I'm wondering if there is some equivalent to bash notifying you of new mail in /var/mail/$USER. Bash does this with MAILCHECK and MAILPATH.

Thanks.

**Comment by faho** [comment]
Fish doesn't have anything to do with mail. This isn't a feature that it has, and I don't believe we want to add it because it's weird to have your shell handle it.

This is a historical feature that was added because it was convenient decades ago and there's really nothing that makes it a part of the shell's responsibility except that it was added back then.

You can write a thing that checks for mail and call it e.g. in your prompt.

**Reply by faho** [reply, marked as answer]
Or build it yourself:

```fish
function checkmail --on-event fish_prompt
    if set -q mailtime
        set -l newtime (path mtime $MAILPATH)
        if test $newtime -gt $mailtime
            echo NEW MAIL
            set -g mailtime $newtime
        end
    else
        set -g mailtime (path mtime $MAILPATH)
    end
end
```

**Reply by koozz** [reply]
Nice answer, this made me come up with:

```fish
function mailcheck --on-event fish_prompt
  if mail -e -f /var/mail/$USER
    set -l count (grep -c "^From " /var/mail/$USER)
    echo "✉️ You've got $count mail"
  end
end
```
